Added a type parameter to the button setup constructor.

PhilippRedel opened the pull request:
I added a type parameter to the button constructer and scaleform to account for the different types.
The actual function of the different types must be researched further, this is just the base setup for that.

Types: 
0 = Standard Button with list select on the right
1 = Standard Button
2 = Brighter Button (maybe disabled state)
3 = Button with 5 small white rectangles on the right for showing a value
4 = Flashing Button (colors fade from white to black endless)

![image](https://user-images.githubusercontent.com/16289288/126370602-e4cc394b-259c-4e4b-a911-7c0299a4b249.png)

PhilippRedel committed:
Added a type parameter to the button setup constructor.

CritteRo commented:
Huh. This is an interesting one. Although while the type var is a good addition, I think we need new events for different types of buttons, instead of craming it in AddButton.

This also needs to be researched further, since we need to check what `GetPauseMenuSelection()` will return, for list items. (currently it returns menuID and uniqueID of the button.) since this is what I use to actually check what the player uses.

PhilippRedel commented:
Yeah im absolutely on your site with this, i will add different events for the different button types.
And of course it needs a lot more research that im currently working on so if there is something new i will add it to the pull request and document it if that is okay for you.

CritteRo committed:
Added comments to button type and button initialIndex

initial Index is basically a background pill for the rightText. 0 = disabled. 1 and up are colors.
